lib/samples/navigate_route_with_rerouting/navigate_route_with_rerouting.dart

```diff
@@ -0,0 +1,532 @@
+        // Get the next direction from the route's direction maneuvers.
+        var nextDirection = '';
+        final directionManeuvers =
+            status.routeResult.routes.first.directionManeuvers;
+        final nextManeuverIndex = status.currentManeuverIndex + 1;
+        if (nextManeuverIndex < directionManeuvers.length) {
+          nextDirection = directionManeuvers[nextManeuverIndex].directionText;
+        }
+
+        setState(() {
+          _routeStatus = '${status.destinationStatus}';
+          _remainingDistance = '$distanceRemainingText $displayUnit';
+          _remainingTime = timeRemainingText;
+          _nextDirection = nextDirection;
+        });
+
+      case DestinationStatus.reached:
+        if (status.remainingDestinationCount > 1) {
+          setState(() {
+            _routeStatus = 'Intermediate stop reached, continue to next stop.';
+          });
+          await _routeTracker.switchToNextDestination();
+        } else {
+          await stop();
+          await reset();
+          setState(() {
+            _routeStatus = 'Destination reached.';
+          });
+        }
+    }
+  }
+
+  // Start the navigation.
+  Future<void> start() async {
+    _mapViewController.locationDisplay.autoPanMode =
+        LocationDisplayAutoPanMode.navigation;
+    await _mapViewController.locationDisplay.dataSource.start();
+    setState(() => _isNavigating = true);
+  }
+
+  // Recenter the map view on the current location.
+  void recenter() {
+    _mapViewController.locationDisplay.autoPanMode =
+        LocationDisplayAutoPanMode.navigation;
+  }
+
+  // Stop the navigation.
+  Future<void> stop() async {
+    // Stop the location display.
+    _mapViewController.locationDisplay.autoPanMode =
+        LocationDisplayAutoPanMode.off;
+    await _mapViewController.locationDisplay.dataSource.stop();
+    setState(() => _isNavigating = false);
+  }
+
+  // Reset the navigation to begin again.
+  Future<void> reset() async {
+    _traversedRouteBuilder =
+        PolylineBuilder(spatialReference: SpatialReference.wgs84);
+    _routeTraveledGraphic.geometry = null;
+    _remainingRouteGraphic.geometry = _routeResult.routes.first.routeGeometry;
+
+    setState(() {
+      _isNavigating = false;
+      _needRecenter = false;
+      _routeStatus = '';
+      _remainingDistance = '';
+      _remainingTime = '';
+      _nextDirection = '';
+    });
+
+    await initNavigation();
+  }
+
+  // Create a simulated location data source.
+  Future<SimulatedLocationDataSource> getLocationDataSource() async {
+    final routePointJson =
+        await rootBundle.loadString('assets/SanDiegoTourPath.json');
+    final routeLine = Geometry.fromJsonString(routePointJson) as Polyline;
+
+    final simulatedLocationDataSource = SimulatedLocationDataSource()
+      ..setLocationsWithPolyline(
+        routeLine,
+        simulationParameters: SimulationParameters(
+          speed: 40,
```

Comment:
I can see why you've increased the speed, but I think it's a little too quick for the built up area and also that the voice isn't as effective since there are deliberately so many instructions / recalculations right at the start. Maybe go with 20 so it's a little quicker than the default but still feels less rushed.

Reply:
`40` is from design, I changed it to 35,  which Koltin sample also use the number.

Reply:
Oh yeah so it is - I did review the design while I was thinking about this but for some reason didn't see it before, interesting. Ok 35 is fine then - hopefully there aren't too many speed cameras 😄